src/icalendar/cal.py

```diff
@@ -17,9 +17,13 @@
 from icalendar.timezone_cache import _timezone_cache
 
 import pytz
+import six
 import dateutil.rrule
 from pytz.tzinfo import DstTzInfo
 
+if six.PY3:
+    unicode = str
```

Comment:
I need some time to look at your definitions in compat. Since you suggested to use six.PY3, I thought this is fine. I adapted the test (put some accentuated characters in it, to force encoding problems) and also corrected cal.py to use `unicode` (happened while refactoring)